scripts/o3de/o3de/download.py

```diff
@@ -124,9 +124,11 @@ def download_o3de_object(object_name: str, default_folder_name: str, dest_path:
     if not dest_path:
         logger.error(f'Destination path cannot be empty.')
         return 1
-    if dest_path.exists():
+    if dest_path.exists() and not force_overwrite:
         logger.error(f'Destination path {dest_path} already exists.')
         return 1
+    elif dest_path.exists():
+        shutil.rmtree(dest_path)
```

Comment:
Personally I think the logic flow is a little odd here I would instead check if force_overwrite is true in the scope of `if dest_path.exists()` evaluating to true. That way we only have one check if dest_path.exists().